From a2445c964a34682977b335098905f248661af004 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 11 Jan 2022 16:08:40 +0800
Subject: [PATCH 02/54] Revert "gl-renderer: Don't require buffer age when
 using partial update"

This reverts commit a2a8d382e38fdee65adb91ce7b4e6fb280389c8f.

This would cause blinking with MALI.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/renderer-gl/gl-renderer.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/libweston/renderer-gl/gl-renderer.c b/libweston/renderer-gl/gl-renderer.c
index a5f5eae..7231f73 100644
--- a/libweston/renderer-gl/gl-renderer.c
+++ b/libweston/renderer-gl/gl-renderer.c
@@ -1404,7 +1404,7 @@ output_get_damage(struct weston_output *output,
 	EGLBoolean ret;
 	int i;
 
-	if (gr->has_egl_buffer_age || gr->has_egl_partial_update) {
+	if (gr->has_egl_buffer_age) {
 		ret = eglQuerySurface(gr->egl_display, go->egl_surface,
 				      EGL_BUFFER_AGE_EXT, &buffer_age);
 		if (ret == EGL_FALSE) {
@@ -1444,7 +1444,7 @@ output_rotate_damage(struct weston_output *output,
 	struct gl_output_state *go = get_output_state(output);
 	struct gl_renderer *gr = get_renderer(output->compositor);
 
-	if (!gr->has_egl_buffer_age && !gr->has_egl_partial_update)
+	if (!gr->has_egl_buffer_age)
 		return;
 
 	go->buffer_damage_index += BUFFER_DAMAGE_COUNT - 1;
-- 
2.20.1

